apidef/oas/schema/x-tyk-api-gateway.json

```diff
@@ -2256,9 +2256,74 @@
           ]
         }
       },
-      "required": [
-        "enabled"
-      ]
+      "required": ["enabled"]
+    },
+    "X-Tyk-TLSTransport": {
+      "type": "object",
+      "properties": {
+        "insecureSkipVerify": {
+          "type": "boolean"
+        },
+        "ciphers": {
+          "type": "array",
+          "items": {
+            "type": "string",
+            "enum": [
+              "TLS_RSA_WITH_RC4_128_SHA",
+              "TLS_RSA_WITH_3DES_EDE_CBC_SHA",
+              "TLS_RSA_WITH_AES_128_CBC_SHA",
+              "TLS_RSA_WITH_AES_256_CBC_SHA",
+              "TLS_RSA_WITH_AES_128_CBC_SHA256",
+              "TLS_RSA_WITH_AES_128_GCM_SHA256",
+              "TLS_RSA_WITH_AES_256_GCM_SHA384",
+              "TLS_ECDHE_ECDSA_WITH_RC4_128_SHA",
+              "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA",
+              "TLS_ECDHE_ECDSA_WITH_AES_256_CBC_SHA",
+              "TLS_ECDHE_RSA_WITH_RC4_128_SHA",
+              "TLS_ECDHE_RSA_WITH_3DES_EDE_CBC_SHA",
+              "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA",
+              "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA",
+              "TLS_ECDHE_ECDSA_WITH_AES_128_CBC_SHA256",
+              "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256",
+              "TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256",
+              "TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256",
+              "TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384",
+              "TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384",
+              "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305_SHA256",
+              "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305_SHA256",
+              "TLS_AES_128_GCM_SHA256",
+              "TLS_AES_256_GCM_SHA384",
+              "TLS_CHACHA20_POLY1305_SHA256",
+              "TLS_FALLBACK_SCSV",
+              "TLS_ECDHE_RSA_WITH_CHACHA20_POLY1305",
+              "TLS_ECDHE_ECDSA_WITH_CHACHA20_POLY1305"
```

Comment:
Ciphers get added, removed, is there a way to not maintain a definite list which will be bitrot?
E.g. link to an implementation documentation page (crypto tls, other).

Reply:
i used the list defined here https://github.com/TykTechnologies/tyk/blob/b6fe04b9d3b7c8a6693df30b8e3ebfc3e891ef44/gateway/cert.go#L44-L67
Best thing to do in this case is remove the json schema validation and rely on the runtime validation here